coding_style/README.adoc

```diff
@@ -150,6 +150,16 @@ Rationale:: https://yaml.org/type/bool.html[YAML 1.1] allows all variants wherea
   But do use quotes for user-side strings such as descriptions, names, and messages.
 * Even if JSON is valid YAML and Ansible understands it, do only use JSON syntax if it makes sense (e.g. a variable file automatically generated) or adds to the readability.
   In doubt, nobody expects JSON so stick to YAML.
+* A single jinja template should not exceed 250 lines in length
+* The aggregate number of jinja includes should not exceed 3
+* Jinja templates should not be used to create structured data but instead text and semi-structured data
```

Comment:
Why not, I'm pretty sure, I've already done this and didn't feel especially bad about it.

Reply:
Agreed.

Reply:
While it can be done, Python provides far more tools for creating and manipulating data than a templating language.  Complex Jinja statements are especially challenging for less experienced users and are not especially readable.

Recent example:
I was helping someone troubleshoot a playbook they were working on.  They wanted to build a dict that used vars for some keys and some values.  In Python this sort of thing is trivial.  But since they were doing it in a playbook I had to walk them through understanding how to 1. build a dict in Jinja in the first place 2. write the template so that it would go through the Jinja parser before the YaML parser 3. how to include both vars and string literals where there were multiple special characters involved.  And this wasn't even that complicated of an example.